packages/world/src/utils/RecipeUtils.sol

```diff
@@ -5,33 +5,57 @@ import { Recipes, RecipesData } from "../codegen/tables/Recipes.sol";
 
 import { ObjectTypeId, NullObjectTypeId } from "../ObjectTypeIds.sol";
 
+struct Recipe {
+  ObjectTypeId stationObjectTypeId;
+  ObjectTypeId[] inputTypes;
+  uint16[] inputAmounts;
+  ObjectTypeId[] outputTypes;
+  uint16[] outputAmounts;
+}
+
+function createRecipe(
+  ObjectTypeId stationObjectTypeId,
+  ObjectTypeId[] memory inputTypes,
+  uint16[] memory inputAmounts,
+  ObjectTypeId[] memory outputTypes,
+  uint16[] memory outputAmounts
+) {
+  bytes32 recipeId = hashInputs(stationObjectTypeId, inputTypes, inputAmounts);
+
+  uint16[] memory _outputTypes;
```

Comment:
Could you explain this bit of solidity? Not sure I understand whats happening here. Is it cuz you dont want to loop over and unwrap?

Reply:
yeah it's just casting the array without having to copy everything